File: 2020/Day6/QuestionCounter.cs
using System;
using System.Linq;
using System.IO;
using System.Text.RegularExpressions;
using System.Collections.Generic;

class QuestionCounter
{
    public static void Main(string[] args)
    {
        string input = File.ReadAllText("input.txt");
        string[] groups = Regex.Split(input, @"\r\n\r\n");

        int finalQuestionCount = 0;

        foreach(var group in groups) {
            string[] people = group.Split(new[] {"\r\n", "\r", "\n"}, StringSplitOptions.None);
            string allAnswers = string.Join("", people);

            int countOfQuests = 0;

            foreach (var character in allAnswers.ToCharArray())
            {
                if(allAnswers.Split(character).Length - 1 == people.Length) {
                    countOfQuests++;
                    allAnswers = allAnswers.Replace(character.ToString(), "");
                }
            }
            finalQuestionCount += countOfQuests;
        }

        Console.WriteLine("\n==== Found " + finalQuestionCount + " matching Questions. ====");
    }
}

//Only in script mode
QuestionCounter.Main(null); 

File: 2020/Day6/QuestionCounter.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Unable to read file");
    let groups: Vec<&str> = input.split("\r\n\r\n").collect();

    let mut count: usize = 0;

    for group in groups {
        let mut questions: Vec<char> = Vec::new();

        for quest in group.chars() {
            if !questions.contains(&quest) && quest != '\n' && quest != '\r' {
                questions.push(quest);
            }
        }
        count += questions.len();
    }
    println!("Counted {} questions.", count);
}